pkg/machinery/resources/network/ethernet_spec.go

```diff
@@ -0,0 +1,72 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package network
+
+import (
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/meta"
+	"github.com/cosi-project/runtime/pkg/resource/protobuf"
+	"github.com/cosi-project/runtime/pkg/resource/typed"
+
+	"github.com/siderolabs/talos/pkg/machinery/proto"
+)
+
+// EthernetSpecType is type of EthernetSpec resource.
+const EthernetSpecType = resource.Type("EthernetSpecs.net.talos.dev")
+
+// EthernetSpec resource holds Ethernet network link status.
+type EthernetSpec = typed.Resource[EthernetSpecSpec, EthernetSpecExtension]
+
+// EthernetSpecSpec describes config of Ethernet link.
+//
+//gotagsrewrite:gen
+type EthernetSpecSpec struct {
+	Rings EthernetRingsSpec `yaml:"rings,omitempty" protobuf:"1"`
+}
+
+// EthernetRingsSpec describes config of Ethernet rings.
+//
+//gotagsrewrite:gen
+type EthernetRingsSpec struct {
+	RX           *uint32 `yaml:"rx,omitempty" protobuf:"1"`
+	TX           *uint32 `yaml:"tx,omitempty" protobuf:"4"`
+	RXMini       *uint32 `yaml:"rx-mini,omitempty" protobuf:"2"`
+	RXJumbo      *uint32 `yaml:"rx-jumbo,omitempty" protobuf:"3"`
+	RXBufLen     *uint32 `yaml:"rx-buf-len,omitempty" protobuf:"5"`
```

Comment:
nit, the order of protobuf fields tx, then rxbuflen seems pretty weird. no practical implications though

Reply:
I tried to follow partially the order in `ethtool` and Linux netlink, but no particular order (fine to change it).